contractcourt/anchor_resolver.go

```diff
@@ -97,12 +100,16 @@ func (c *anchorResolver) Resolve() (ContractResolver, error) {
 	// to the sweeper.
 	relayFeeRate := c.Sweeper.RelayFeePerKW()
 
+	witnessType := input.CommitmentAnchor
+
+	// For taproot channels, we need to use the proper witness type.
+	if c.chanType.IsTaproot() {
```

Comment:
> . A future design/refactor is to put witnessType inside the sign descriptor, 

IMO the sign descriptor is just for signing, it describes which key to use, etc -- but is insufficient to create an entire witness. I don't the control block path here (putting the control block in the sign desc) was the _best_ route, but I liked it as it meant callers to the sign methods didn't need to create/parse/include the control block themselves. Instead those that prepped the sign desc handled that, while the signers could just call sign like normal (eg: the sweeper, etc).

Reply:
Inspired by the new layer `ScriptDescriptor`, I think the design can be built around witness types and channel types. afaict, `StandardWitnessType` is the lowest layer that defines all possible spending paths, in which contains the info about the channel types. Then for each witness type, we have a sign descriptor that deals with creating the sig based on the witness. The witness type itself also tells us the channel type because it has a many-to-one relationship with channel types. Maybe that's an improved route, again something for future endeavor 🦾